src/wallet/wallet.cpp

```diff
@@ -4713,6 +4713,31 @@ void CWallet::GetScriptForMining(std::shared_ptr<CReserveScript> &script)
     script->reserveScript = CScript() << ToByteVector(pubkey) << OP_CHECKSIG;
 }
 
+void CWallet::GetScriptForMining(std::shared_ptr<CReserveScript> &script)
+{
+    std::string sAddress = gArgs.GetArg("-miningaddress", "");
+    if (!sAddress.empty()) {
+        CTxDestination dest = DecodeDestination(sAddress);
+
+        if (IsValidDestination(dest)) {
+            if (IsMine(dest) != ISMINE_SPENDABLE) {
+                WalletLogPrintf("-miningaddress %s not under my control, falling back to random address", sAddress);
+                return GetScriptForMiningReserveKey(script);
+            }
+            // Disallow Stealth Addresses for now
+            CBitcoinAddress address(sAddress);
+            if (address.IsValidStealthAddress()) {
+                WalletLogPrintf("-miningaddress must be a basecoin address");
+                return GetScriptForMiningReserveKey(script);
+            }
```

Comment:
This is unnecessary since this check is being done in veild.cpp and qt/veil.cpp

Reply:
It's also being done in rpc/mining.cpp--I presume as a backstop in case the arg was changed between startup and the rpc to start mining. If it's changed, and manages to get by here, the script will be empty, and I think that will make it fail when it tries to commit a block (I tried mining to stealth on testnet at some point and iirc that's what happened). Perhaps that is user error, since we should not be mutating miningaddress in-process? The difference is that we can't return an RPC error here, and I felt it was better to fall back to existing behavior (mining to an unexpected address) rather than surprising behavior (mining fails at the last possible moment).

Reply:
Cleared these checks out.